src/tmpfiles/tmpfiles.c

```diff
@@ -1291,20 +1298,25 @@ static int write_one_file(Item *i, const char *path) {
 }
 
 typedef int (*action_t)(Item *, const char *);
-typedef int (*fdaction_t)(Item *, int fd, const struct stat *st);
+typedef int (*fdaction_t)(Item *, int fd);
 
-static int item_do(Item *i, int fd, const struct stat *st, fdaction_t action) {
+static int item_do(Item *i, int fd, fdaction_t action) {
+        struct stat st;
         int r = 0, q;
 
         assert(i);
         assert(fd >= 0);
-        assert(st);
 
         /* This returns the first error we run into, but nevertheless
          * tries to go on */
-        r = action(i, fd, st);
+        r = action(i, fd);
 
-        if (S_ISDIR(st->st_mode)) {
+        if (fstat(fd, &st) < 0) {
+                r = r ?: -errno;
```

Comment:
Downgrade-to-bool is frowned upon. Better to write this as
```C
if (r >= 0)
     r = -errno;
```

Reply:
ok